src/ddsrt/include/dds/ddsrt/ifaddrs.h

```diff
@@ -36,15 +43,38 @@ struct ddsrt_ifaddrs {
 
 typedef struct ddsrt_ifaddrs ddsrt_ifaddrs_t;
 
+/**
+ * @brief Get the interface addresses (@ref ddsrt_ifaddrs) for specific address families
+ * 
+ * Note that array 'afs' must be terminated with an element 'DDSRT_AF_TERM'.
+ * 
+ * @param[out] ifap interface address pointer
+ * @param[in] afs an array of address families
+ * @return a DDS_RETCODE
```

Comment:
And the possible return codes are quite limited: `OK`, `NOT_ALLOWED`, `OUT_OF_RESOURCES` or the catch-all `ERROR`.